python/lsst/ts/scheduler/driver/feature_scheduler.py

```diff
@@ -0,0 +1,443 @@
+        survey_topology: `lsst.ts.scheduler.kernel.SurveyTopology`
+
+        Raises
+        ------
+        RuntimeError:
+            If `config` does not have a `scheduler_config` attribute or if it
+            points to a non-existing file.
+
+        """
+
+        if not hasattr(config, "scheduler_config"):
+            raise RuntimeError("No feature scheduler configuration defined.")
+        elif not os.path.exists(config.scheduler_config):
+            raise RuntimeError(
+                f"Feature scheduler configuration file {config.scheduler_config} not found."
+            )
+
+        if self.scheduler is None or config.get("force", False):
+            self.log.info(
+                f"Loading feature based scheduler configuration from {config.scheduler_config}."
+            )
+            spec = importlib.util.spec_from_file_location(
+                "config", config.scheduler_config
+            )
+            conf = importlib.util.module_from_spec(spec)
+            spec.loader.exec_module(conf)
+            if not hasattr(conf, "scheduler"):
+                raise NoSchedulerError("No scheduler defined in configuration file.")
+            self.scheduler = conf.scheduler
+            if not hasattr(conf, "nside"):
+                raise NoNsideError("No nside defined in configuration file.")
+            self.nside = conf.nside
+            self.conditions = Conditions(nside=self.nside)
+
+            self.conditions.FWHMeff = dict(
+                [
+                    (key, np.empty(hp.nside2npix(self.nside), dtype=float))
+                    for key in self.models["seeing"].filter_list
+                ]
+            )
+
+        # self.scheduler.survey_lists is a list of lists with different surveys
+        survey_names = [
+            survey.survey_name
+            for survey_list in self.scheduler.survey_lists
+            for survey in survey_list
+        ]
+
+        survey_topology = SurveyTopology()
+
+        survey_topology.num_general_props = len(survey_names)
+        survey_topology.general_propos = survey_names
+        survey_topology.num_seq_props = 0
+        survey_topology.sequence_propos = []
+
+        return survey_topology
+
+    def cold_start(self, observations):
+        """Rebuilds the internal state of the scheduler from a list of
+        observations.
+
+        Parameters
+        ----------
+        observations : list of Observation objects
+
+        """
+        raise RuntimeError("Cold start not supported by SequentialScheduler.")
+
+    def update_conditions(self):
+        """Update conditions on the scheduler.
+        """
+
+        super().update_conditions()
+
+        self._format_conditions()
+
+        # Update conditions on the scheduler
+        self.scheduler.update_conditions(self.conditions)
+
+        # Set time from next observation based on the current time on the
+        # observatory model, which accounts for current observations on the
+        # queue.
```

Comment:
Sorry for nitpicking on your English but shouldn't this be "time to next observation"? Or does this code add another observation right after the next one, because in that case the sentence is correct as it is now.